pyvista/core/_vtk_core.py

```diff
@@ -507,3 +507,46 @@ def VTKVersionInfo():
 
 
 vtk_version_info = VTKVersionInfo()
+
+
+def override(vtk_class):
+    """Override a VTK class with a PyVista class.
+
+    Returns
+    -------
+    callable
+        Decorator to override the VTK class with a PyVista class.
+
+    """
+
+    def decorator(cls):
+        if vtk_version_info >= (9, 4):
+            vtk_class.override(cls)
+        return cls
+
+    return decorator
+
+
+class vtkPyVistaOverride:
+    """Base class to automatically override VTK classes with PyVista classes."""
+
+    # TODO: `MutableSequence` base class is breaking the VTK override mechanism
+    VTK_OVERRIDE_EXCLUSIONS = (
+        'vtkMultiBlockDataSet',
+        'vtkPartitionedDataSet',
+    )
```

Comment:
I think we need to raise an upstream issue for when `MutableSequence` is in the base classes it breaks the override mechanism in https://github.com/Kitware/VTK/blob/master/Wrapping/PythonCore/PyVTKObject.cxx

Reply:
Posted upstream in https://gitlab.kitware.com/vtk/vtk/-/issues/19545

Reply:
For these we should just call `.override(None)` to unregister the parent VTK classes

Reply:
A fix is inbound here: https://gitlab.kitware.com/vtk/vtk/-/merge_requests/11698